proposals/044-integration-model.md

```diff
@@ -0,0 +1,203 @@
+
+## Integration Model
+
+* Issue: https://github.com/syndesisio/syndesis-project/issues/22
+* Sprint: 14
+* Affected Repos:
+  - syndesis-rest
+  - syndesis-ui
+
+## Objective
+
+Currently integrations that are deployed are completely disconnected from the Syndesis DB afterwards. There is no way yet to update a running integration nor to see the state of the currently running integration.
+
+The purpose of this document is to suggest a concise state model for `Integrations` as well as define the interaction between the `Integration`s designed in _syndesis-ui_ with the integrations running in the OpenShift cluster whose creation is triggered by a deployment action by the user.
+
+When rethinking the integration and the state model, the following points should be considered:
+
+* Multiple versions of an Integration must be able to exist.
+* Exactly one of these integrations can be active.
+* Each deployment of an Integration creates a new version.
+* Users can switch integration versions from the UI
+* Versioned Integrations can be inspected from a history page.
+* Old integrations can't be changed but can be used as a blueprint for new integrations.
+* As soon as an integration is changed, a new version of an integration is created. All changes go to this new integration until it is deployed by the user. After this, its gets a fixed version and cannot be changed anymore.
+* The user must always be able to see and examine the currently active integration along with its version number.
+
+_Attention: THIS IS WORK-IN-PROGRESS. The propsoal is PRed early to collect early feedback_
+
+
+## Integration state model
+
+Technical an Integration is the domain object describing a Camel Route. 
+It is managed by `syndesis-rest` and `syndesis-ui`. 
+During design time this object only lives in this domain. 
```

Comment:
This section is a bit unclear. Actually I mean it lives only in the DB but there is not already an associated runtime (pod with camel routs) created.

I would distinguish between "Save" (store it in the DB, lightweight operation) and "Publish" (create the runtime deployment, heavyweight operation).

A "Publish" simply means then to change the `targetState` from "Draft" to "Active" in the DB stored integration. A background controller in syndesis-rest would then pick this up and create the pods. 

So imo the UI is still free to save either explicitely or implicitely for the 'save' operation. Only the "Publish" operation should be a dedicated step (== button).